### ISSUE TITLE
copy-dbalogin slow when there are many users in the databases.

### ISSUE THREAD

**Issue by klindberg70** [dataplat/dbatools#8879]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

takes long time

### Steps to Reproduce

Copy-DbaLogin -source sql301\tdb050 -Destination vm02-1073-d\SQLI01_1159_t -Login andpal 

### Please confirm that you are running the most recent version of dbatools

1.1.118

### Other details or mentions

Copy-DbaLogin -source sql301\tdb050 -Destination vm02-1073-d\SQLI01_1159_t -Login andpal takes about one minute.

VERBOSE: [16:23:44][Update-SqlPermission] Grant CONNECT SQL to andpal on vm02-1073-d\SQLI01_1159_T successfully performed.
VERBOSE: [16:24:40][Update-SqlPermission] Database [xxx] is not accessible on SQL301\TDB050. Skipping.

there is an extreme amount of users (3211 ) in the database and the command is hanging on 
INSERT #loginmappings select suser_sname(u.sid), db_name(), u.name, null from sys.database_principals AS u where suser_sname(u.sid) is not null for 30 sec. (output 798 rows) wait info PREEMPTIVE_OS_LOOKUPACCOUNTSID

### What PowerShell host was used when producing this error

Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version

Name                           Value                                                                                                                                                                         
----                           -----                                                                                                                                                                         
PSVersion                      5.1.17763.3770                                                                                                                                                                
PSEdition                      Desktop                                                                                                                                                                       
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                       
BuildVersion                   10.0.17763.3770                                                                                                                                                               
CLRVersion                     4.0.30319.42000                                                                                                                                                               
WSManStackVersion              3.0                                                                                                                                                                           
PSRemotingProtocolVersion      2.3                                                                                                                                                                           
SerializationVersion           1.1.0.1                                                                                                                                                                       

### SQL Server Edition and Build number

Microsoft SQL Server 2016 (SP3-CU1-GDR) (KB5021128) - 13.0.7024.30 (X64)   Dec 24 2022 20:17:51   Copyright (c) Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2012 R2 Standard 6.3 <X64> (Build 9600: ) (Hypervisor) 

Microsoft SQL Server 2019 (RTM-CU19) (KB5023049) - 15.0.4298.1 (X64)   Jan 27 2023 16:44:09   Copyright (C) 2019 Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2019 Datacenter 10.0 <X64> (Build 17763: ) (Hypervisor) 

### .NET Framework Version

.NET Framework 4.8.4614.0

**Comment by andreasjordan** [comment]
We don't run the sql code, the code is here:
https://github.com/microsoft/sqlmanagementobjects/blob/d6f3e68fe6fae09d22355a160baa525d8032fd64/src/Microsoft/SqlServer/Management/SqlEnum/xml/DatabaseMappings.xml#L7

We only use the method EnumDatabaseMappings to get the mappings. So I think there is nothing we can do here.

**Comment by klindberg70** [comment]
Would it be possible to have an option to skip the control of the mappings to the databases?

**Comment by wsmelton** [comment]
You can do the copy process using other commands in the module. Copy commands are not able to cover every environment scenario, I've had many that I had to roll my own setup to cover a number of things specific to customer's conditions and configurations.

**Comment by klindberg70** [comment]
Ok it was not obvious to me how to copy using other commands. Also tried export-dbalogin with the same issue.
Ended up using sp_help_revlogin and as another workaround I could copy the logins before the database was present in the new instance. I do not see the need for checking the database user mapping when I copy the logins.

**Comment by klindberg70** [comment]
Yes that works! of course why didn't I find that one. Thanks a lot for your attention and a great job

**Comment by andreasjordan** [comment]
Thanks for the feedback!
